src/store/sagas/list.js

```diff
@@ -0,0 +1,92 @@
+import {
+  take,
+  takeEvery,
+  takeLatest,
+  call,
+  all,
+  put,
+} from "redux-saga/effects";
+import {
+  ALL_LIST,
+  ALL_LIST_SAGA,
+  DELETE_LIST,
+  DELETE_LIST_SAGA,
+  EDIT_LIST,
+  EDIT_LIST_SAGA,
+  ADD_LIST,
+  ADD_LIST_SAGA,
+} from "../types";
+import axios from "axios";
+const wait = (ms) => new Promise((res) => setTimeout(res, ms));
+
+//list
+export function* allList(actions) {
+  let url = "http://localhost:3001/list/all";
+  const setHeaders = { headers: { "Content-Type": "application/json" } };
+  let res = yield fetch(url, setHeaders);
+  res = yield res.json();
+  console.log(res);
+  yield put({ type: ALL_LIST, payload: { lists: res } });
+}
+
+//add
+export function* addList(actions) {
+  yield wait(1000);
```

Comment:
why manual wait here?

Reply:
Sir when user click many times on button with in 1 second, and after stop it will take the only latest item & only one item will be added to database.